### PR TITLE
Blog - How to use Docker in VS Code

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@jalletto 
Please review

**Comment by msali123** [comment]
@jalletto

**Comment by msali123** [comment]
@adamgordonbell

**Comment by msali123** [comment]
@jalletto  Done with replacement of GIF's.

**Comment by jalletto** [comment]
Preview: https://6360debf35097500b2106710--ephemeral-narwhal-3ac3b2.netlify.app/blog/how-to-use-docker-in-vscode/

**Review by jalletto — CHANGES_REQUESTED** [review]

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+### Viewing Container Logs
+
+This option is available in the context menu for running [containers](/blog/docker-slim). The integrated terminal will display the logs.
+
+Using the running Node.js container as our example, all you have to do is navigate to Docker Explorer. In the Containers tab, right-click on your container and choose View Logs. You should see it being displayed in the terminal
+
+![view container logs]({{site.images}}{{page.slug}}/P8lZ9Fr.jpeg)\
+
+![container logs]({{site.images}}{{page.slug}}/Hn6x6xm.png)\
```
[the remark this answers is not in the document]
Comment (jalletto) [reply blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:162]: I don't see them in the images folder. Maybe you forgot to update these?

**Review by jalletto — CHANGES_REQUESTED** [review]

**Comment by msali123** [comment]
@jalletto my bad didn't verified .

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+## Prerequisites
+
+To follow along with this article, you will need to have [Docker](/blog/rails-with-docker) installed on your workstation. Instructions on installing and running Docker are [available](https://docs.docker.com/get-docker/), and they should be specific to the operating system you are running. You also need to have [VS Code installed](https://code.visualstudio.com/download).
+
+## Installing the Docker Extension
+
+Head to the extension section in VS Code and type `docker` in the search box. You should see something like this:
+
+![Searching for docker extension]({{site.images}}{{page.slug}}/vVdGELj.png)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:28]:
```suggestion
![Searching for docker extension]({{site.images}}{{page.slug}}/vVdGELj.png)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+The first selection in this image is what you want to look for to install. Click on it to install. Once it's done downloading, you will notice a Docker icon or logo at the bottom left corner of your window. This is the [Docker](/blog/rails-with-docker) explorer.
+
+![Docker icon in VS Code]({{site.images}}{{page.slug}}/wqmhmlw.gif)\
+
+Click on it. All our running and stopped containers are highlighted inside the Docker explorer. Here, you can also see your images, registries, volumes, networks, e.t.c. :
+
+![containers,volumes,images and more]({{site.images}}{{page.slug}}/UQWjQRM.png)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:36]:
```suggestion
![Containers ,volumes, images and more]({{site.images}}{{page.slug}}/UQWjQRM.png)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+If you have existing projects with Docker or docker-compose files or you simply prefer to write your own, don't worry, the extension still has a lot to offer you. But if you do want to speed up creating these files, you have the option of letting the extension do some of the heavy-lifting for you.
+
+To generate the Docker files automatically, open the Command Palette by pressing `⇧⌘P` on a Mac or `Control+Shift+P` on a Windows PC. Then type:
+
+~~~{.bash caption=""}
+Docker: Add Docker files to Workspace command`:
+~~~
+
+![adding docker files]({{site.images}}{{page.slug}}/bM7WWsn.jpeg)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:81]:
```suggestion
![Adding docker files]({{site.images}}{{page.slug}}/bM7WWsn.jpeg)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+**USER:**Sets the user name or UID to use when running the image in addition to any subsequent CMD, ENTRYPOINT, or RUN instructions that follow it in the Dockerfile.
+**CMD:** Provides defaults for an executing container.
+
+Similar to the `.gitignore`, the `.dockerignore` instructs Docker to hold files and folders that should not be replicated when creating the image.
+
+## Building and Running Your Docker File
+
+To build the Docker image, open the `Command Palette` and execute `Docker Images: Build Image`. You can also right-click the `Dockerfile` in the navigation panel and select `Build image:`
+
+![Docker build]({{site.images}}{{page.slug}}/gn9rNt4.jpeg)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:118]:
```suggestion
![Building an image]({{site.images}}{{page.slug}}/gn9rNt4.jpeg)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+If you check the extension pane and look at the `images` section inside the Docker explorer, you should see the latest project has been added to the docker explorer.
+
+![our latest project]({{site.images}}{{page.slug}}/qxnlDbj.png)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:122]:
```suggestion
![Our latest project]({{site.images}}{{page.slug}}/qxnlDbj.png)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+The following step is to run our `image`. Open the command palette once more, type `docker run`, and then pick `Docker: Run`. It will display a list of all the containers on your system. Select the `docker-node:latest` tag and click `Enter`.
+  
+![Docker run]({{site.images}}{{page.slug}}/lDRhXAu.jpeg)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:126]:
```suggestion
![Running a container]({{site.images}}{{page.slug}}/lDRhXAu.jpeg)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+You can also run the container by going to the left pane, selecting the Docker explorer, then under `IMAGES`, choose the image you want. Right-click on  `latest`. and click run. You will get the same logs running on the terminal.
+
+![running docker inside the extension]({{site.images}}{{page.slug}}/OU6RnwE.jpeg)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:130]:
```suggestion
![Running docker inside the extension]({{site.images}}{{page.slug}}/OU6RnwE.jpeg)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+You can view the app running in the container in the browser. To do this, right-click on the running container in the docker explorer and click on "open in browser":
+
+ ![view in browser]({{site.images}}{{page.slug}}/Em7k5gV.jpeg)\
+
+### Debugging Our Container
+
+The docker extension includes a VS Code debugger configuration inside `.vscode/launch.json` for debugging when running inside a container. To do this, set a breakpoint in the `get()` handler for the '/' in `index.js` by pressing the f9 key. Then go to the `run and debug section` in VS Code and select `Docker Node.js launch` debugger and start debugging by pressing `f5` :
+
+![the run and debug section]({{site.images}}{{page.slug}}/WSiiCWl.png)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:144]:
```suggestion
![The run and debug section]({{site.images}}{{page.slug}}/WSiiCWl.png)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+
+![selecting docker node.js launch]({{site.images}}{{page.slug}}/jcR4gOc.png)\
+
+What you'll notice is that the debugger comes to a halt in `index.js` at the breakpoint:
+
+![The breakpoint]({{site.images}}{{page.slug}}/dnqMcyX.png)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:150]:
```suggestion
![Using breakpoints]({{site.images}}{{page.slug}}/dnqMcyX.png)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+You can then continue by pressing the play button at the top to continue running.
+
+### Viewing Container Logs
+
+This option is available in the context menu for running [containers](/blog/docker-slim). The integrated terminal will display the logs.
+
+Using the running Node.js container as our example, all you have to do is navigate to Docker Explorer. In the Containers tab, right-click on your container and choose View Logs. You should see it being displayed in the terminal
+
+![view container logs]({{site.images}}{{page.slug}}/P8lZ9Fr.jpeg)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:160]:
```suggestion
![Viewing container logs]({{site.images}}{{page.slug}}/P8lZ9Fr.jpeg)
```

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+
+![container logs]({{site.images}}{{page.slug}}/Hn6x6xm.png)\
+
+### Docker Inspect Images
+
+The Docker inspect images is a feature that allows you to inspect the images built and see the details in a JSON file. This allows you to see important information about our image i.e the image ID, when it was created, volumes, and many more. Inside our docker explorer, navigate to `IMAGES` and locate the `project folder/latest` right click and click on inspect:
+
+![Docker inspect]({{site.images}}{{page.slug}}/HMQtXXA.jpeg)\
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:168]:
```suggestion
![Docker inspect]({{site.images}}{{page.slug}}/HMQtXXA.jpeg)
```

**Review by jalletto — APPROVED** [review]

File: blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+![Searching for docker extension]({{site.images}}{{page.slug}}/vVdGELj.png)
+
+The first selection in this image is what you want to look for to install. Click on it to install. Once it's done downloading, you will notice a Docker icon or logo at the bottom left corner of your window. This is the [Docker](/blog/rails-with-docker) explorer.
+
+![Docker icon in VS Code]({{site.images}}{{page.slug}}/wqmhmlw.gif)\
+
+Click on it. All our running and stopped containers are highlighted inside the Docker explorer. Here, you can also see your images, registries, volumes, networks, e.t.c. :
+
+![Containers ,volumes, images and more]({{site.images}}{{page.slug}}/UQWjQRM.png)
```
Comment (jalletto) [inline blog/_posts/2022-10-31-how-to-use-docker-in-vscode.md:36]:
```suggestion
![Containers ,volumes, images, and more]({{site.images}}{{page.slug}}/UQWjQRM.png)
```

**Pull request merged**